README.md

```diff
@@ -18,3 +18,82 @@ Module documentation is [published on Pursuit](http://pursuit.purescript.org/pac
 
 You can find an example usage [here](test/Main.purs).
 
+## `Lens` vs `ALens`: a Note on Impredicativity
+
+Sometimes, you may encounter some strange errors to do with types that
+_should_ unify but don't. For example:
+
+```purescript
+newtype Wrapped = Wrapped String
+derive instance newtypeWrapped :: Newtype Wrapped _
+
+-- `_Newtype` is an `Iso'`, so we can specialise to `Lens'`.
+good :: Lens' Wrapped String
+good = _Newtype
+
+-- So, we should be able to do the same when lifted into a record.
+type Example = { lens :: Iso' Wrapped String }
+
+
+bad :: Example -> Lens' Wrapped String
+bad = _.lens
+
+-- Could not match constrained type
+--    Profunctor t2 => t2 String String -> t2 Wrapped Wrapped
+--  with type
+--    p0 String String -> p0 Wrapped Wrapped
+```
+
+The subtle issue here relates to a subject called *impredictivaty*. We can make
```

Comment:
You have a typo here!

Reply:
Haha, noo! Doesn't fill anyone with confidence, right?